flexget/entry.py

```diff
@@ -301,3 +334,35 @@ def __hash__(self):
 
     def __repr__(self):
         return '<Entry(title=%s,state=%s)>' % (self['title'], self._state)
+
+
+lazy_func_registry = {}
+
+
+class LazyFunc:
+    def __init__(self, lazy_func_name, fields, plugin=None):
+        self.name = lazy_func_name
+        self.fields = fields
+        self.plugin = plugin
+        self._func = None
+
+    def __call__(self, func):
+        self._func = func
+        lazy_func_registry[self.name] = self
+        return func
+
+    @property
+    def function(self):
+        if '.' in self._func.__qualname__:
```

Comment:
maybe something in the `inspect` module can help with something more deterministic than this, or do you think that is robust enough?

Reply:
I think it's probably fine. I'd say let's wait until we find an example where it causes a problem to change it.